amr-wind/physics/SuperEllipseBody.cpp

```diff
@@ -0,0 +1,102 @@
+#include "amr-wind/physics/SuperEllipseBody.H"
+#include "amr-wind/CFDSim.H"
+#include "AMReX_iMultiFab.H"
+#include "AMReX_MultiFabUtil.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_ParReduce.H"
+#include "amr-wind/utilities/trig_ops.H"
+#include "amr-wind/utilities/IOManager.H"
+#include "amr-wind/utilities/io_utils.H"
+#include "amr-wind/utilities/linear_interpolation.H"
+
+namespace amr_wind::superellipsebody {
+
+namespace {
+
+//! Utility function to parse inputs and return a vector instance
+inline vs::Vector parse_vector(amrex::ParmParse& pp, const std::string& key)
+{
+    amrex::Vector<amrex::Real> tmp;
+    pp.getarr(key.data(), tmp);
+    AMREX_ALWAYS_ASSERT(tmp.size() == 3U);
+
+    return vs::Vector{tmp[0], tmp[1], tmp[2]};
+}
+
+
+} // namespace
+
+SuperEllipseBody::SuperEllipseBody(CFDSim& sim)
+    : m_sim(sim)
+    , m_repo(sim.repo())
+    , m_mesh(sim.mesh())
+    , m_body_blank(sim.repo().declare_int_field("terrain_blank", 1, 1, 1))
+    , m_terrain_drag(sim.repo().declare_int_field("terrain_drag", 1, 1, 1))
+    , m_terrainz0(sim.repo().declare_field("terrainz0", 1, 1, 1))
+{
+
+    amrex::ParmParse pp(identifier());
+    pp.query("body_file", m_body_file);
+    m_dim = parse_vector(pp, "dimensions");
+
+    m_sim.io_manager().register_output_int_var("terrain_blank");
+    m_sim.io_manager().register_output_int_var("terrain_drag");
+    m_sim.io_manager().register_io_var("terrainz0");
+    m_body_blank.setVal(0.0);
+    m_terrain_drag.setVal(0.0);
+    m_terrainz0.setVal(1e-4);
```

Comment:
Is 1e-4 meaningful or just meant to be a small number? If the latter, let's use something in `constants`